konnectivity-client/pkg/client/client_test.go

```diff
@@ -59,6 +59,49 @@ func TestDial(t *testing.T) {
 	}
 }
 
+func TestDialRace(t *testing.T) {
```

Comment:
Maybe we want to detect go routines leaks in this test case? You can do something like this PR is doing with the uber pkg: https://github.com/kubernetes-sigs/apiserver-network-proxy/pull/336

Note the use of `VerifyNone()` as that ignores the goroutine that is running due to klog.

Reply:
Given that the original problem being solved was a goroutine leak, would be nice to add this. Not sure this test is likely to catch such a leak but the wrapper seems low cost.